cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/GitHubHttpHandler.java

```diff
@@ -0,0 +1,453 @@
+    if (gitStore.getRepo(repo) != null) {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(gitStore.getRepo(repo)));
+    } else {
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
+    }
+  }
+
+  @Path("repos/github/{repo}")
+  @PUT
+  public void addRepoInfo(FullHttpRequest request, HttpResponder responder,
+                          @NotNull @PathParam("repo") String repo) throws Exception {
+
+      GitHubRepo githubRequest = GSON.fromJson(request.content().toString(StandardCharsets.UTF_8)
+          , GitHubRepo.class);
+
+      if (githubRequest.validateAllFields()) {
+        //test the repo's connection to ensure it exists.
+        int responseCode = testRepoConnection(githubRequest,
+            "token " + githubRequest.getAuthString());
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+          String authString = "token " + githubRequest.getAuthString();
+          SecureKeyId authKeyId = new SecureKeyId("system", repo + "_auth_token");
+          secureStoreManager.put("system", repo + "_auth_token", authString,
+              repo + "Authorization Token", new HashMap<String, String>());
+          gitStore.addOrUpdateRepo(repo, githubRequest.getUrl(), githubRequest.getDefaultBranch(),
+              repo + "_auth_token");
+          responder.sendString(HttpResponseStatus.OK, "Repository Information Saved.");
+        } else if (responseCode == HttpURLConnection.HTTP_MOVED_PERM) {
+          responder.sendString(HttpResponseStatus.MOVED_PERMANENTLY, "Repository has been moved.");
+        } else if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
+          responder.sendString(HttpResponseStatus.FORBIDDEN, "You do not have access to this repository.");
+        } else {
+          responder.sendString(HttpResponseStatus.NOT_FOUND, "Repository was not found.");
+        }
+        //If the user leaves fields blank, throw back which fields are missing
+      } else {
+        String errorString = "Please enter a ";
+        if (!githubRequest.validNickname()) {
+          errorString += "nickname, ";
+        }
+        if (!githubRequest.validUrl()) {
+          errorString += "url, ";
+        }
+        if (!githubRequest.validDefaultBranch()) {
+          errorString += "default branch, ";
+        }
+        if (!githubRequest.validAuthString()) {
+          errorString += "authorization token, ";
+        }
+        responder.sendString(HttpResponseStatus.BAD_REQUEST,
+            errorString.substring(0, errorString.length() - 2));
+      }
+  }
+
+  /**
+   * Deletes current repository info
+   */
+  @Path("repos/github/{repo}")
+  @DELETE
+  public void deleteRepoInfo(HttpRequest request, HttpResponder responder,
+                              @NotNull @PathParam("repo") String repo) throws Exception {
+    if (gitStore.getRepo(repo) != null) {
+      gitStore.deleteRepo(repo);
+      secureStoreManager.delete("system", repo + "_auth_token");
```

Comment:
same here, use the static final field instead of hardcode the "_auth_token" string here.